### ISSUE TITLE
Pylint does not respect false values for boolean config options in toml files

### ISSUE THREAD

**Issue by GrimzEcho** [pylint-dev/pylint#8460]
### Bug description

When pylint loads its configuration from a .toml file, it treats the mere presence of a key as enabling that flag or option, without regard to its actual value.

For instance, setting `from-stdin = false` in a toml config will cause pylint to attempt to read from stdin. Setting `exit-zero = false` will cause pylint to always exit with code 0 even if there are errors or warnings.

The only way to disable the flag is to comment out the line. This is directly at odds with the example configuration provided in the documentation (https://pylint.readthedocs.io/en/latest/user_guide/configuration/all-options.html) which has many examples of `option = false`.

This same behavior seems to happen with _pylintrc_ configuration files as well, but I only did some basic testing.

### Configuration

```toml
# example 1:
[tool.pylint.main]
from-stdin = false
exit-zero = false

# example 2:
[tool.pylint.main]
# from-stdin = false
exit-zero = false

# example 3:
[tool.pylint.main]
# from-stdin = false
# exit-zero = false
```

### Command used

```shell
pylint test.py; echo $?
```
```python
# test.py
import datetime
```

### Pylint output

```shell
# example 1
# hangs because it is expecting input from stdin ... must be killed with ctrl-c


# example 2
************* Module test
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:1:0: W0611: Unused import datetime (unused-import)
------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
0


# example 3
************* Module test
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:1:0: W0611: Unused import datetime (unused-import)
------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
20
```

### Expected behavior

Examples 1 and 2 should produce the same output as example 3

### Pylint version

```shell
pylint 2.17.0
astroid 2.15.0
Python 3.7.3 (default, Oct 31 2022, 14:04:00) 
[GCC 8.3.0]
```

### OS / Environment

Debian 10

### Additional dependencies

None

**Comment by DanielNoord** [comment]
This is an issue with `store_true` options. Basically, these options don't work with `false` or `true` but act as flags.

We should indeed fix this and also update the documentation. Thanks for spotting it!

**Comment by ianrose1** [comment]
hello, can I try and work on this issue?

**Comment by Pierre-Sassoulas** [comment]
Sure thing @ianrose1, I assigned you to the issue, thank you !

**Comment by ianrose1** [comment]
Hey, just wanted to let you know my current findings, and see if I'm on the right path. I was able to reproduce the bug locally, and I've been looking at some of the functions in pylint/pylint/config/argument.py and pylint/pylint/config/argument_manager.py. I see how the store_true class is initialized, but I'm confused about how to make a change to fix the bug. Any help would be appreciated, thank you!

**Comment by DanielNoord** [comment]
I think the issue is in this piece of code:
https://github.com/pylint-dev/pylint/blob/bc9c5c6610cce2a068eabce598200399b9d002f2/pylint/config/config_file_parser.py#L79-L82

What we currently do is add ``--{config}`` if we find a `store_true` `config` in the `toml` file. However, this doesn't take into account that if the value is actually `False` to should be ``--no-{config}`` to make sure the flag is negated.

**Comment by ianrose1** [comment]
Got it, I'll work on that
